### PR TITLE
Bmereu tema2

### PR DESCRIPTION
**Pull request by 914-Mereu-Bianca**

### CODE REVIEW

File: repository/Repository.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Repository::Repository() {}
+
+Repository::~Repository() {}
+
+void Repository::add(Product const& p){
+    this->v.push_back(p);
+}
+
+void Repository::remove(Product const& p){
+    for(auto it = this->v.begin(); it != this->v.end(); it++){
```
Comment (CristianIonutIonescu) [inline repository/Repository.cpp:13]: minor:
Use pre-increment operator ( ++it) as it's better optimised.
The compiler already does it for you behind the scenes but it's a good practice to write it this way

File: repository/Repository.cpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#include "Repository.h"
+#include <string>
+
+Repository::Repository() {}
```
Comment (CristianIonutIonescu) [inline repository/Repository.cpp:4]: you could write both as =default

File: model/Product.h [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+{
+private:
+    std::string name;
+    int weight;
+public:
+    Product(std::string _name, int _weight);
+    //copy constructor
+    Product(const Product & p)=default;
+    //move constructor 
+    Product(Product&& p)=default;
```
Comment (CristianIonutIonescu) [inline model/Product.h:16]: missing move operator

File: repository/Repository.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+class Repository
+{
+private:
+    std::vector<Product> v;
+    std::string file;
+public:
+    Repository();
+    Repository(std::string const& _file);
+    Repository(Repository & r)=default;
+    Repository(Repository && r)=default;
```
Comment (CristianIonutIonescu) [inline repository/Repository.h:15]: missing move/copy operators to have Rule of 5 complete

File: main.cpp [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#include "model/Product.h"
+#include "repository/Repository.h"
+#include <iostream>
+
+using namespace std;
```
Comment (CristianIonutIonescu) [inline main.cpp:5]: please don't include the whole std namespace in your project. It's bad practice

**Review by CristianIonutIonescu — CHANGES_REQUESTED** [review]
Hello,

Good job with the PR. I have left some minor comments on it but overall is looking great.

File: repository/Repository.h [hunk 6]
```diff
@@ -0,0 +1,24 @@
+#ifndef _PROGRAM_REPOSITORY_REPOSITORY_H_
+#define _PROGRAM_REPOSITORY_REPOSITORY_H_
+#include <vector>
+#include "/home/bianca/2022/model/Product.h"
+
+class Repository
+{
+private:
+    std::vector<Product> v;
```
Comment (enachebogdan) [inline repository/Repository.h:9]: Surely you can find a better name than "v" for your member

**Review by CristianIonutIonescu — APPROVED** [review]

**Pull request merged**